components/CopyWithOneClick.js

```diff
@@ -0,0 +1,36 @@
+import { useState } from "react";
+import { Icon } from "@iconify/react";
+import styled from "styled-components";
+
+export default function CopyWithOneClick({ hasIconText, iconColor, codeData }) {
+  const [isCopied, setIsCopied] = useState(false);
+  async function handleClick() {
+    setIsCopied(true);
+    setTimeout(() => {
+      setIsCopied(false);
+    }, 3000);
+    await navigator.clipboard.writeText(codeData);
+  }
+
+  return (
+    <StyledButton type="button" onClick={handleClick}>
+      <Icon
+        icon={isCopied === true ? "mingcute:check-fill" : "fa-regular:copy"}
+        color={iconColor}
+        aria-label={isCopied === true ? "code copied" : "copy code"}
+      />
+      {hasIconText && (isCopied ? "code copied" : "copy code")}
+    </StyledButton>
+  );
+}
+
+const StyledButton = styled.button`
+  position: absolute;
+  background: transparent;
+  border: none;
+  display: flex;
+  align-items: center;
+  gap: 0.3rem;
+  right: 1rem;
+  bottom: 1rem;
+`;
```

Comment:
Make sure to position your button so that the code block does not cover the button :)
See the example in the Card "Test 8732"